parasoft-findings-teamcity-agent/src/main/resources/com/parasoft/findings/teamcity/agent/sa-pmdcpd.xsl

```diff
@@ -48,14 +48,54 @@
             </xsl:if>
             <xsl:if test="@srcRngFile">
                 <xsl:attribute name="path">
-                    <xsl:if test="../../@lang = 'java'">
-                        <xsl:value-of select="substring-after(@srcRngFile, /ResultsSession/@prjModule)"/>
-                    </xsl:if>
-                    <xsl:if test="../../@lang != 'java'">
-                        <xsl:value-of select="@srcRngFile"/>
-                    </xsl:if>
+                    <xsl:choose>
+                        <xsl:when test="not(/ResultsSession/@prjModule)">
+                            <xsl:apply-templates select="/ResultsSession/CodingStandards/Projects/Project">
+                                <xsl:with-param name="srcRngFile" select="@srcRngFile"/>
+                            </xsl:apply-templates>
+                        </xsl:when>
+                        <xsl:otherwise>
+                            <xsl:apply-templates select="/ResultsSession/Scope/Locations/Loc">
+                                <xsl:with-param name="projectLocRef" select="@locRef"/>
+                                <xsl:with-param name="srcRngFile" select="@srcRngFile"/>
+                            </xsl:apply-templates>
+                        </xsl:otherwise>
+                    </xsl:choose>
                 </xsl:attribute>
             </xsl:if>
         </xsl:element>
     </xsl:template>
+
+    <xsl:template match="Project">
+        <xsl:param name="srcRngFile"/>
+        <xsl:param name="resProjPath"/>
+        <xsl:if test="$srcRngFile and contains($srcRngFile, concat(@name,'/', @name))">
```

Comment:
Let's separate the special handling for c++ professional and dottest into different templates for better code readability. 
Please put some comments as explanations.

Reply:
So, at line 53 and line 87, instead of using "apply-templates", create different templates and use "call-template".